Server-side rendering safety & sensible defaults

jackdclark opened the pull request:
Resolves #31 

* Put guards around the usage of the `window` object to prevent uncaught exceptions when rendering on the server
* Provide sensible defaults when rendered on the server

lpww commented:
How come we didn't need to do anything for `geolocation`, `media-controls`, and `mouse-position`?

lpww commented on src/hooks/scroll.js:
```diff
@@ -1,9 +1,10 @@
   const [pos, setPos] = useState({
-    top: window.scrollY,
-    left: window.scrollX
+    top: IS_SERVER ? 0 : window.scrollY,
```
I noticed that you didn't check for `window` in the `useEffect` callback. Is that because `window` is defined by the time it is called? If so, could we avoid this check by always defaulting to `0` here (or null) and then updating the state to the true values in `useEffect`? 

The media-controls hook had to use that approach to set initial values as soon as the `ref` is [rendered](https://github.com/nearform/react-browser-hooks/blob/master/src/hooks/media-controls.js#L56:L59) but [not before](https://github.com/nearform/react-browser-hooks/blob/master/src/hooks/media-controls.js#L4:L8). It works well and I noticed that it didn't need changes for SSR, so maybe that's a pattern we can adopt to support SSR without needing these specific checks?

jackdclark replied:
`useEffect` doesn't run on the server, just when it mounts in the DOM (same as `componentDidMount`), so we're safe to access `window` there.

The pattern of using static defaults and then updating state with `window` values within `useEffect` will work but it will introduce an extra render as the state will always update once it's mounted. What do you think?

lpww replied:
That's a good point. It's probably more important to avoid unnecessary renders. I'll merge this as it is

lpww replied:
Any chance you can bump package.json?

lpww commented:
Is it worth adding something in the docs to mention that our hooks all support SSR?

lpww commented:
I was planning to merge this pr but a test started failing

jackdclark commented:
@lpww
- updated the `useOnline` tests to account for the changes in #109  
- added a note to the README regarding SSR support

This should be good to go now.
